### PR TITLE
Core/Scripts: zone_shattrath_city cleanup

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**
- restore missing text lines for Sal'salabim and Raliq the Drunk
- replace hardcoded text with gossip added in the database
- move gossip ID, quest ID, spell ID (and so on) into enum

**Target branches:** 3.3.5 + 6.x

**Issues addressed**:
This is another script cleanup in preparation for future SAI conversions.
- updates https://github.com/velinath/TrinityCore/issues/2 (remove hardcoded text)
- added missing "say attack" text lines to Sal'salabim and Raliq the Drunk
- general cleanup by moving "magic numbers" into enum lists

**Tests performed**: (Does it build? Tested in-game?) Yes and yes.
Tested with 2 different characters as quest takers / players.

**Known issues and TODO list:**
Edit: these 2 gossip options are now in working order, properly linked via `action_menu_id` .
- [X] `Who are the Sha'tar?` `(7735,0 ->7736,9458)`
- [X] `Isn't Shattrath a draenei city?  Why do you allow others here?` `(7735,1 ->7737,9459)`

`gossip_menu` `7735` (NPC default) had to be used for both `gossip_menu_option`.`id` 0 and 1.
After inserting the gossip options this way, they had to be linked to `gossip_menu`.`entry` 7736 & 7737
in order to open `gossip_menu`.`text_id` 9458 + 9459 (= `npc_text`.`ID` ). This method can be used
even if the sub-gossip does not have any menu options, only gossip text. I did not know that until now. :smile:

### CODE REVIEW

File: src/server/scripts/Outland/zone_shattrath_city.cpp [hunk 1]
```diff
@@ -471,8 +518,8 @@ class npc_ishanah : public CreatureScript
         if (creature->IsQuestGiver())
             player->PrepareQuestMenu(creature->GetGUID());
 
-        player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, ISANAH_GOSSIP_1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
-        player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, ISANAH_GOSSIP_2, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+2);
+        player->ADD_GOSSIP_ITEM_DB(MENU_OPTION_WHO_ARE_THE_SHATAR, OPTION_ID_WHO_ARE_THE_SHATAR, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
+        player->ADD_GOSSIP_ITEM_DB(MENU_OPTION_WHY_DO_YOU_ALLOW,   OPTION_ID_WHY_DO_YOU_ALLOW,   GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+2);
```
Comment (joschiwald) [inline src/server/scripts/Outland/zone_shattrath_city.cpp]: this is doable with gossip_* tables only, no conditions or sai, no need for cpp script

Comment (tkrokli) [inline src/server/scripts/Outland/zone_shattrath_city.cpp]: Good point, thanks for pointing them out to me. Will make an update ASAP.

**Comment by tkrokli** [comment]
Well, I even tried to change `gossip_menu_option` like this, but it did not seem to help for me:
```sql
DELETE FROM `gossip_menu_option` WHERE `menu_id` IN (7735,7736,7737);
INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`OptionBroadcastTextID`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`,`BoxBroadcastTextID`) VALUES
(7735,0,0,"Who are the Sha'tar?",                                          15642,1,1,0,0,0,0,'',0),
(7735,1,0,"Isn't Shattrath a draenei city?  Why do you allow others here?",15644,1,1,0,0,0,0,'',0);
```
Tried changing other details as well, like changing the contents in the table `gossip_menu`, but no luck:
```sql
DELETE FROM `gossip_menu` WHERE `entry` = 7735;
INSERT INTO `gossip_menu` (`entry`,`text_id`) VALUES
(7735,9457),
(7735,9458),
(7735,9459);
```
I still can't get those 2 gossip options to show up as they used to do before. What am I doing wrong?

**Comment by ccrs** [comment]
assign the gossip menu id  in creature_template and check if it has the correct flag

**Comment by tkrokli** [comment]
```sql
SELECT `entry`,`name`,`subname`,`gossip_menu_id`,`npcflag` FROM `creature_template`
WHERE `entry` = 18538;
```
```
entry  name     subname                      gossip_menu_id  npcflag
-----  -------  ---------------------------  --------------  ---------
18538  Ishanah  High Priestess of the Aldor            7735        3
```

**Pull request closed without merging**